format_time.rb

```diff
@@ -0,0 +1,35 @@
+class FormatTime
+
+  ACCEPTABLE_FORMATS = %w[year month day hour minute second].freeze
+  TIME_FORMATS = {
+    year: '%Y',
+    month: '%m',
+    day: '%d',
+    hour: '%H',
+    minute: '%M',
+    second: '%S'
+  }.freeze
+
+  def initialize(formats)
+    @formats = formats
+  end
+
+  def valid?
+    invalid_formats.empty?
+  end
+
+  def time
+    time_output = valid_formats.map { |format| TIME_FORMATS[format.to_sym] }
+    Time.now.strftime(time_output.join('-'))
+  end
+
+  def invalid_formats
+    @formats - valid_formats
+  end
+
+  private
+
+  def valid_formats
+    @formats.select { |format| ACCEPTABLE_FORMATS.include?(format) }
```

Comment:
```suggestion
    @formats.select { |format| TIME_FORMATS.key?(format) }    
```
Это даже быстрее будет работать, чем массив